src/shogun/preprocessor/RandomFourierGaussPreproc.cpp

```diff
@@ -8,393 +8,100 @@
-		int32_t *dim_input_space2, float64_t* kernelwidth2) const {
-
-	ASSERT(randomcoeff_additive2)
-	ASSERT(randomcoeff_multiplicative2)
-
-	if (!test_rfinited()) {
-		*dim_feature_space2 = 0;
-		*dim_input_space2 = 0;
-		*kernelwidth2=1;
-		*randomcoeff_additive2 = NULL;
-		*randomcoeff_multiplicative2 = NULL;
-		return;
-	}
-
-	*dim_feature_space2 = cur_dim_feature_space;
-	*dim_input_space2 = cur_dim_input_space;
-	*kernelwidth2=cur_kernelwidth;
-
-	*randomcoeff_additive2 = SG_MALLOC(float64_t, cur_dim_feature_space);
-	*randomcoeff_multiplicative2 = SG_MALLOC(float64_t, cur_dim_feature_space*cur_dim_input_space);
-
-	std::copy(randomcoeff_additive, randomcoeff_additive+cur_dim_feature_space,
-			*randomcoeff_additive2);
-	std::copy(randomcoeff_multiplicative, randomcoeff_multiplicative+cur_dim_feature_space*cur_dim_input_space,
-			*randomcoeff_multiplicative2);
-
-
-}
-
-void RandomFourierGaussPreproc::set_randomcoefficients(
-		float64_t *randomcoeff_additive2,
-		float64_t * randomcoeff_multiplicative2,
-		const int32_t dim_feature_space2, const int32_t dim_input_space2, const float64_t kernelwidth2) {
-	dim_feature_space = dim_feature_space2;
-	dim_input_space = dim_input_space2;
-	kernelwidth=kernelwidth2;
-
-	SG_FREE(randomcoeff_multiplicative);
-	randomcoeff_multiplicative=NULL;
-	SG_FREE(randomcoeff_additive);
-	randomcoeff_additive=NULL;
-
-	cur_dim_feature_space=dim_feature_space;
-	cur_dim_input_space = dim_input_space;
-	cur_kernelwidth=kernelwidth;
-
-	if( (dim_feature_space>0) && (dim_input_space>0) )
-	{
-	randomcoeff_additive=SG_MALLOC(float64_t, cur_dim_feature_space);
-	randomcoeff_multiplicative=SG_MALLOC(float64_t, cur_dim_feature_space*cur_dim_input_space);
-
-	std::copy(randomcoeff_additive2, randomcoeff_additive2
-			+ dim_feature_space, randomcoeff_additive);
-	std::copy(randomcoeff_multiplicative2, randomcoeff_multiplicative2
-			+ cur_dim_feature_space*cur_dim_input_space, randomcoeff_multiplicative);
-	}
+	UniformRealDistribution<float64_t> uniform(0.0, 2.0 * Math::PI);
 
+	for (auto row : range(m_dim_output))
+		m_offset[row] = uniform(m_prng);
 }
 
 void RandomFourierGaussPreproc::fit(std::shared_ptr<Features> f)
 {
-	if (dim_feature_space <= 0) {
-		throw ShogunException(
-				"CRandomFourierGaussPreproc::init (Features *f): dim_feature_space<=0 is not allowed, use void set_dim_feature_space(const int32 dim) before!\n");
-	}
-
-	io::info("calling CRandomFourierGaussPreproc::init(...)");
-	int32_t num_features =
-	    f->as<DenseFeatures<float64_t>>()->get_num_features();
-
-	if (!test_rfinited()) {
-		dim_input_space = num_features;
-		init_randomcoefficients();
-		ASSERT( test_rfinited())
-	} else {
-		dim_input_space = num_features;
-		// does not reinit if dimension is the same to avoid overriding a previous call of set_randomcoefficients(...)
-		init_randomcoefficients();
-	}
+	auto num_features = f->as<DenseFeatures<float64_t>>()->get_num_features();
```

Comment:
it kinda sucks that we need to fix the float64 type here in order to get the dimensionality of the feature space :(
But the method doesnt exist in the base classes for some reason ...

Reply:
@gf712 any thoughts about this?

Reply:
Is there a reason why this is not implemented in Features?

Reply:
I guess because for string features, it doesnt make sense ...

Reply:
But then this should be implemented in DotFeatures rather than DenseFeatures?

Reply:
I think some dot features also can have varying dimensionality

Reply:
I guess this means there should be a base class for `Features` that have a known number of features..